scripts/data/src/org/akvo/gae/remoteapi/CorrectSurveyedLocale.java

```diff
@@ -0,0 +1,195 @@
+ * survyedLocaleIdentifier
+ */
+public class CorrectSurveyedLocale implements Process {
+
+    @Override
+    public void execute(DatastoreService ds, String[] args) throws Exception {
+        if (args.length == 0) {
+            System.err
+                    .println("Usage: "
+                            + RemoteAPI.class.getName()
+                            + "CorrectSurveyedLocale <appid> <username> <password> <appid> <surveyId>");
+            System.exit(1);
+        }
+
+        // retrieve registration form id for surveygroup matching monitored group id passed in
+        Long surveyId = Long.parseLong(args[0]);
+        Entity monitoringSurvey = retrieveMonitoringSurvey(ds, surveyId);
+        List<Entity> surveyForms = retrieveSurveyForms(ds, surveyId);
+
+        // map locale identifiers and surveyedlocale ids for all survey instances created by
+        // registration form
+        Long registrationFormId = (Long) monitoringSurvey.getProperty("newLocaleSurveyId");
+
+        if (registrationFormId == null) {
+            System.err.println("Unable to identify registration form.");
+            System.exit(1);
+        }
+
+        // retrieve all monitoring surveys for survey group
+        List<Entity> dataPoints = retrieveSurveyInstances(ds, registrationFormId);
+        Map<String, Long> surveyedLocaleDeviceIdentifierMap = mapDeviceIdentifierSurveyedLocaleId(dataPoints);
+
+        // for each monitoring survey
+        for (Entity monitoringForm : surveyForms) {
+            Long formId = new Long(monitoringForm.getKey().getId());
+            if (registrationFormId.equals(formId)) {
+                continue; // skip registration form processing
+            }
+
+            List<Entity> monitoringFormInstances = retrieveSurveyInstances(ds, formId);
+            List<Entity> correctedMonitoringFormInstances = correctSurveyedLocale(
+                    monitoringFormInstances, surveyedLocaleDeviceIdentifierMap, ds);
+            batchSaveEntities(ds, correctedMonitoringFormInstances);
+        }
+    }
+
+    /**
+     * Retrieve the monitoring survey entity
+     *
+     * @param ds
+     * @param surveyId
+     * @return
+     * @throws EntityNotFoundException
+     */
+    private Entity retrieveMonitoringSurvey(DatastoreService ds, Long surveyId)
+            throws EntityNotFoundException {
+        return ds.get(KeyFactory.createKey(SURVEY_KIND, surveyId));
+    }
+
+    /**
+     * Retrieve all forms belonging to the monitoring survey
+     *
+     * @param ds
+     * @param surveyId
+     * @return
+     */
+    private List<Entity> retrieveSurveyForms(DatastoreService ds, Long surveyId) {
+        Filter surveyFilter = new Query.FilterPredicate("surveyGroupId", FilterOperator.EQUAL,
+                surveyId);
+        Query q = new Query(FORM_KIND).setFilter(surveyFilter);
+
+        List<Entity> surveyForms = new ArrayList<Entity>();
+        for (Entity survey : ds.prepare(q).asIterable()) {
```

Comment:
Since the list of forms is not huge, we can just do: `ds.prepare(q).asList()` instead of getting an `Iterable` and adding it to a `List` ?